packages/app/src/pages/CreateReturn.tsx

```diff
@@ -0,0 +1,173 @@
+
+  const orderId = params?.orderId
+  const goBackUrl =
+    orderId != null
+      ? appRoutes.details.makePath(orderId)
+      : appRoutes.home.makePath()
+
+  const { order, isLoading, mutateOrder } = useOrderDetails(orderId ?? '')
+
+  const { sdkClient } = useCoreSdkProvider()
+  const { data: createdReturnObj } = useCoreApi(
+    'returns',
+    'create',
+    returnNeedsCreation
+      ? [
+          {
+            order: sdkClient.orders.relationship(order.id)
+          },
+          {
+            include: ['origin_address', 'destination_address']
+          }
+        ]
+      : null
+  )
+
+  useEffect(() => {
+    if (!isLoading) {
+      const draftReturnObject = order.returns?.filter(
+        (returnObj) =>
+          returnObj.status === 'draft' &&
+          (returnObj.return_line_items ?? []).length === 0
+      )[0]
+      if (draftReturnObject != null) {
+        setReturnObj(draftReturnObject)
+      } else {
+        setReturnNeedsCreation(true)
+      }
+    }
+  }, [order, isLoading])
+
+  useEffect(() => {
+    if (createdReturnObj != null) {
+      setReturnObj(createdReturnObj)
+    }
+  }, [createdReturnObj])
+
+  const returnableLineItems = useReturnableList(order)
+  const {
+    createReturnLineItemsError,
+    createReturnLineItems,
+    isCreatingReturnLineItems
+  } = useCreateReturnLineItems()
+
+  if (
+    returnObj == null ||
+    order.fulfillment_status !== 'fulfilled' ||
+    returnableLineItems.length === 0 ||
+    !canUser('create', 'returns')
+  ) {
+    return (
+      <PageLayout
+        title='Request return'
+        onGoBack={() => {
+          setLocation(goBackUrl)
+        }}
+      >
+        <EmptyState
+          title='Permission Denied'
+          description='You cannot create a return for this order or you are not authorized to access this page.'
+          action={
+            <Link href={goBackUrl}>
+              <Button variant='primary'>Go back</Button>
+            </Link>
+          }
+        />
+      </PageLayout>
+    )
+  }
+
+  return (
+    <PageLayout
+      title={
+        <SkeletonTemplate isLoading={isLoading}>
+          Request return
+        </SkeletonTemplate>
+      }
+      onGoBack={() => {
+        setLocation(goBackUrl)
+      }}
+    >
+      <ScrollToTop />
+      {returnableLineItems != null && returnableLineItems.length !== 0 && (
+        <>
+          <Spacer bottom='12'>
+            <FormReturn
+              defaultValues={{
+                items: returnableLineItems?.map((item) => ({
+                  quantity: item.quantity,
+                  value: item.id
+                }))
+              }}
+              lineItems={returnableLineItems}
+              apiError={createReturnLineItemsError}
+              onSubmit={(formValues) => {
+                void createReturnLineItems(returnObj, formValues).then(() => {
+                  void mutateOrder()
+                  setLocation(goBackUrl)
```

Comment:
```suggestion
                  void mutateOrder().finally(() => {
                    setLocation(goBackUrl)
                  })
```
To be honest I am not sure if my suggestion is really needed :D 
Maybe just to be sure the order is updated before unmounting the component (change route)

Reply:
Good idea! For sure this would avoid any glitch or visual updating effect when landing back to order detail..